Created data_generation.py

luanaruiz9 opened the pull request:
Created data generation script for the cartesian product problem (original and iterative decoding form).

luanaruiz9 committed:
Created data_generation.py

Created data generation script for the cartesian product problem (original and iterative decoding form).

luanaruiz9 committed:
Corrected row case of iteratively_decode()

santiaontanon-google commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,277 @@
+                             repeat_digits = False, reverse = False, row = False, 
+                             copy_output = False, short_input = False):
+  """
+  Adapted from CompositionalExperimentsTransformers-EMNLP2021-submission.ipynb
```
I think this is a Google internal link, which will not be visible externally. So, I think it's ok to skip this :)

santiaontanon-google commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,277 @@
+START_TOKEN = "[START]"
+END_ITERATION_TOKEN = "[ENDITER]"
+
+def create_cartesian_dataset(trainsize, testsize, trainmindigits = 1, 
```
I think default values should be specified without spaces around the "=" operator according to the style guide. Maybe Josh can confirm (here and the rest of the file)

santiaontanon-google commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,277 @@
+      return example_in, [START_TOKEN] + example_out
+
+
+  def iteratively_decode(example_in, example_out, reverse = False, row = False, 
```
This is a long and complex function, a doc string would be great :)

santiaontanon-google commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,277 @@
+                               data_path + "it_dec_train.src",
+                               data_path + "it_dec_train.tgt")
+  # Generating iterative decoding val and test data (easy).
+  ratio = 0.1
```
What is this "ratio" used for? is it to split train/test?

luanaruiz9 replied:
This is the ratio of test samples that will be used for "validation", i.e., to calculate step-by-step accuracy.

santiaontanon-google reviewed (approved):
Thanks Luana, looks good! Maybe just some missing docstrings here and there to clarify what some parts of the code do. Otherwise, it looks good to me!

luanaruiz9 committed:
Updated data_generation.py

To incorporate Santi's suggestions and to include the possibility of having a different quantity of numbers and letters in the hard test samples.

jtainslie commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,312 @@
+# Standard library imports
+import numpy as np
+import numpy.random as random
+# Specify local path in which the data will be saved
```
Nit: it's conventional to have one or two blank lines between the imports and the rest of the code

jtainslie commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,312 @@
+data_path = "MyDrive/preprocess_cartesian/token/data/"
+# Flags
+row = False # whether to predict next row or next pair of tokens
+copy = False # whether to copy previous output to next output
```
Nit: It may help to make these names a bit more descriptive.

jtainslie commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,312 @@
+END_ITERATION_TOKEN = "[ENDITER]"
+
+def create_cartesian_dataset(trainsize, testsize, trainmindigits=1, 
+                             trainmaxdigits=6, testmindigits_nb = 6, 
```
Formatting nit:
```suggestion
                             trainmaxdigits=6, testmindigits_nb=6, 
```

jtainslie commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,312 @@
+                                                                  testmaxdigits_nb,
+                                                                  testmindigits_lt, 
+                                                                  testmaxdigits_lt)
+  return (train_examples, it_dec_train_examples, 
+          test_easy_examples, it_dec_test_easy_examples,
+          test_hard_examples, it_dec_test_hard_examples)
```
Optional nit: when returning so many values in a tuple, it can be helpful instead to create a `dataclass` object to wrap everything in.  This way each value has a descriptive field name instead of relying on tuple ordering.

jtainslie commented on datasets/cartesian/data_generation.py:
```diff
@@ -0,0 +1,312 @@
+                                      testmindigits_lt = 5, testmaxdigits_lt = 6,  
+                                      repeat_digits = False, reverse = False, 
+                                      row = row, copy_output = copy, 
+                                      short_input = short_input)
```
Nit: there should be no spaces around these `=` signs for function arguments like this.

jtainslie reviewed (approved):
Looks good for the most part aside from some nits.  I probably should have mentioned this earlier, but it may be nice to run the code through a formatter like this one: https://github.com/google/yapf

luanaruiz9 committed:
Corrected nits and formatted data_generation.py
